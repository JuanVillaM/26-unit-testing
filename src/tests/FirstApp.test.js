import React from 'react';
import { shallow } from 'enzyme';
import FirstApp from "../FirstApp";

describe('Tests in <FirstApp />', () => {
    test('Must show <FirsyApp /> correctly', () => {
        const greet = 'Hello, I am Goku';
        const wrapper = shallow( <FirstApp greet={ greet } /> );

        expect( wrapper ).toMatchSnapshot();
    })

    test('Must show the subtitle sended by props', () => {
        const greet = 'Hello, I am Goku';
        const subtitle = 'I am a Super Subtitle';
        const wrapper = shallow( <FirstApp greet={ greet } subtitle={ subtitle } /> );
        const paragraphText = wrapper.find('p').text();
        
        expect( paragraphText ).toBe( subtitle );
    })
    
    
});