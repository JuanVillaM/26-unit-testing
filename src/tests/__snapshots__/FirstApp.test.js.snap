// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <FirstApp /> Must show <FirsyApp /> correctly 1`] = `
<Fragment>
  <h1>
    Hello, I am Goku
  </h1>
  <p>
    I am a subtitle.
  </p>
</Fragment>
`;

exports[`Tests in FirstApp.test.js file Must show <FirsyApp /> correctly 1`] = `
<Fragment>
  <h1>
    Hello, I am Goku
  </h1>
  <p>
    I am a subtitle.
  </p>
</Fragment>
`;
